extensions/wikia/WikiaWhiteList/WikiaWhiteList.php

```diff
@@ -93,6 +94,19 @@ function wfWhiteListParse($text) {
 							if (count($m) > 0) {
 								$text = preg_replace(sprintf(REGEX_PARSE_WHITELIST,"rel","nofollow"), "", $text);
 								$text = preg_replace(sprintf(CLASS_PARSE_WHITELIST,"external","(text|free|autonumber)"), "class=\"$1\"", $text);
+
+								$parsed = @parse_url( $lhref );
+								if (!empty( $parsed ) && is_array( $parsed )) {
+									$city_id = WikiFactory::DomainToID(wfNormalizeHost( $parsed['host'] ));
+									if ($city_id) {
+										$parsed['host'] = parse_url(
```

Comment:
this also will need to be updated a little for language wikis. I have a feeling we could use few unit test cases here.